Refactor sequelize setup to use ES6 syntax

henperi opened the pull request:
#### What does this PR do?
- This PR updates the way sequelize is set up so it can easily use es6 syntax.
- This PR also sets up a remote DEV_DB, a local DEV_DB, a remote PROD_DB and a test db to make development a breeze.

#### Description of Tasks completed?
- Update babelrc file
- Create .env and add environment variables for sequelize
- Create a config.js file for sequelize
- Update .sequelizerc to use config.js so as to utilize babel-register
- Install babel-core and babel-register
- Update models/index.js to use proper es6 syntax, and instantiate sequelize properly

#### How should this be manually tested?
- Make a new model and use ES6 syntax, run the migration and watch it succeed.
- If you want to use the remote dev db already set upon just change the NODE_ENV to development.
- If you want to just develop locally on your pc just set NODE_ENV = localDevelopment, then update the MY_POSTICO_URL in .env to use your own local database connection string.

#### What are the relevant pivotal tracker stories?
nill

#### Any background context you want to add?
The production database in the .env file is from Heroku, once we host the app it should begin to work as expected

#### Screenshots
nill

henperi committed:
Refactor sequelize to use ES6 syntax

- Update babelrc file
- Create .env and add environment variables for sequelize
- Create a config.js file for sequelize
- Update .sequelizerc to use config.js so as to utilize babel-register
- Install babel-core and babel-register
- Update models/index.js to use proper es6 syntax, and instantiate sequelize properly

sudhons commented on server/models/index.js:
```diff
@@ -1,19 +1,35 @@
+let DB_URL;
+switch (NODE_ENV) {
+  case 'development':
+    DB_URL = process.env.LOCAL_DATABASE_URL;
```
But `LOCAL_DATABASE_URL` is not defined in the `.env` file.

henperi replied:
Thanks for your observation. I failed to double check, I will make changes as soon as possible.

sudhons commented on server/models/index.js:
```diff
@@ -1,19 +1,35 @@
+    break;
+
+  case 'production':
+    DB_URL = process.env.REMOTE_DATABASE_URL;
```
But REMOTE_DATABASE_URL is not defined in the `.env` file.

henperi replied:
Thanks for your observation. I failed to double check, I will make changes as soon as possible.

sudhons commented on server/models/index.js:
```diff
@@ -1,19 +1,35 @@
+    break;
+
+  case 'test':
+    DB_URL = process.env.TEST_DATABASE_URL;
```
But TEST_DATABASE_URL is not defined in the `.env` file.

henperi replied:
Thanks for your observation. I failed to double check, I will make changes as soon as possible.

sudhons commented on server/models/index.js:
```diff
@@ -1,19 +1,35 @@
+    break;
 }
 
+const sequelize = new Sequelize(DB_URL, DB_URL);
```
Can't `DB_URL` be sent once to the constructor?

henperi replied:
Thanks for observing, parameter two aught to be the options. I left it that way as it still worked since I was using a database connection string which contained the options. Nevertheless, I just made that change. Thanks again.

sudhons commented on .env:
```diff
@@ -1,3 +1,8 @@
-REMOTE_DEV_DB=postgres://dkvnqdsh:HimRBU1zxUYm3Z60-gXlM2-u8lk-iwYp@pellefant.db.elephantsql.com:5432/dkvnqdsh
+NODE_ENV=test
```
It's great that you created a `.env` file but why are you setting `NODE_ENV` in the `.env` file when `.env` usually should not be pushed to github. If `NODE_ENV` is to be set, wouldn't setting it in `package.json` file be better?

henperi replied:
Yeah, thanks for your review. The .env file shouldn't be pushed to GitHub as you've pointed out, but to make sure everyone on our team cloning the project gets up to speed in a short time, I decided to add it here since this is just for practice purposes. 

Setting it in package.json may not be a better option as this environment variable will also need to be set on Heroku. If it's not set, the express application always defaults to use 'development' as it's environment. Cheers.

henperi committed:
Update .env and models/index.js

- Use the correct environmental variable names in server/models/index.js
- set all db connection strings in .env to use elepahnt sql except MY_POSTICO_URL env variable
